services/black/src/ambassador/BlackClientFactory.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using artiso.AdsdHotel.ITOps.Abstraction.NServiceBus;
+using NServiceBus;
+
+namespace artiso.AdsdHotel.Black.Ambassador
+{
+    public class BlackClientFactory
+    {
+        private readonly IHttpClientFactory _httpClientFactory;
+
+        public BlackClientFactory(IHttpClientFactory httpClientFactory)
+        {
+            _httpClientFactory = httpClientFactory;
+        }
+
+        public async Task<BlackClient> CreateAsync()
+        {
+            var endpointConfiguration = NServiceBusEndpointConfigurationFactory.Create(
+                endpointName: $"Black.Client.{Guid.NewGuid()}",
+                rabbitMqConnectionString: "host=localhost");
```

Comment:
We can use the factory, but we should change the factory implementation. Right know it's always with callbacks etc. In the infrastructure project a split these, so we are more flexible.

Reply:
Feel free to implement the flag in the newly-consolidated ITOps project.